### DISCUSSION TITLE
SML certificate update

### DISCUSSION THREAD

**Discussion by sakasaka19** [phax/phoss-smp#271] [Q&A]
Hi Philip

I asked OpenPeppol a question regarding SMP certificate renewal,
I was sent a link to your Technical documentation web site, so let me ask the question directly.

The following explanation is on the screen of PhossSMP.

>The SML will replace the certificate at this date at 02:00am Brussels Time.
>It must be in the future and within the validity period of the provided new public certificate.
>If not provided, the 'valid from' part of the new certificate is used.

Q1 Does it still update at 2:00 as of now?
Q2 Is this a process that takes only a few minutes? Or does it take 30 minutes or something?
Q3 Is it the same process for SMK? (Is it replaced at 2:00?)

Best Regards

"<img width="932" alt="SML Certificate Update" src="https://github.com/phax/phoss-smp/assets/18133370/7cfb07f6-c53b-4929-94b5-c3d0131ad198">"

**Comment by phax** [comment]
Hello @sakasaka19 Thanks for asking :)
Ad Q1: to the best of my knowledge yes. 2:00am CET
Ad Q2: I *think* this is an instant process. As it is only needed for participant registration and deletion, it shouldn't be an issue even if it takes 5 minutes. But I don't know the details - sorry.
Ad Q3: Yes, the logic is the same for SMK and SML. SMK is the same software instance, just with a different configuration.

hth :)

**Reply by sakasaka19** [reply]
Thanks for the answer.
I have another question, if you don't mind.
I tried to update the endpoint certificates for about 4000 participants from phoss-smp GUI.
However, GUI was not responding, perhaps due to the large number of participants.
So, I'm going to update the DB(smp_endpoint) directly.
Is there any problem?

**Reply by phax** [reply]
Did you use the "Bulk update certificate" page to make the changes, or how did you try to do it?
There should be no harm in doing this directly in the DB....

**Reply by sakasaka19** [reply]
I tried updating on the "Bulk update certificate" page, but it seems like I ran out of memory.
I will deal with this by directly updating the DB without using the GUI.
Thank you very much.
